### PR TITLE
Feature/cax 1 prs@212 broker api integration testing

### PR DESCRIPTION
**Pull request by cpeeyush**
- Integration Tests with kafka testcontainer for broker APIs.

### CODE REVIEW

File: coreservices/partsrelationshipservice/Dockerfile [hunk 1]
```diff
@@ -4,10 +4,12 @@ FROM maven:3-jdk-11 AS maven
 WORKDIR /build
 COPY pom.xml .
 COPY integration-tests integration-tests
+COPY borker-proxy-integration-tests borker-proxy-integration-tests
```
Comment (ouphi) [inline coreservices/partsrelationshipservice/Dockerfile:7]:
```suggestion
COPY broker-proxy-integration-tests broker-proxy-integration-tests
```

Comment (ouphi) [reply coreservices/partsrelationshipservice/Dockerfile:7]: I see `borker` is used several times in file names etc.

File: coreservices/partsrelationshipservice/borker-proxy-integration-tests/src/test/java/net/catenax/brokerProxy/UpdatePartsAttributesTest.java [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+    @Test
+    public void updatedPartsAttributes_success() {
+
+        var updateRequest = brokerProxyMother.partAttributeUpdate();
+
+        given()
+                .contentType(ContentType.JSON)
+                .body(updateRequest)
+        .when()
```
Comment (chrislomonico) [inline coreservices/partsrelationshipservice/borker-proxy-integration-tests/src/test/java/net/catenax/brokerProxy/UpdatePartsAttributesTest.java:40]: nit: inconsistent formatting

File: coreservices/partsrelationshipservice/borker-proxy-integration-tests/src/test/java/net/catenax/brokerProxy/UpdatePartsAttributesTest.java [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+                .assertThat()
+                .statusCode(HttpStatus.NO_CONTENT.value());
+
+        assertThat(hasExpectedBrokerEvent(updateRequest, PartAttributeUpdateEvent.class, this::isEqual, configuration.getPartsAttributesTopic())).isTrue();
+    }
+
+    @Test
+    public void updatedPartsAttributesBadRequest_failure() {
+
+                given()
```
Comment (chrislomonico) [inline coreservices/partsrelationshipservice/borker-proxy-integration-tests/src/test/java/net/catenax/brokerProxy/UpdatePartsAttributesTest.java:52]: nit: inconsistent formatting

File: coreservices/partsrelationshipservice/prs-testing/src/main/java/net/catenax/prs/testing/TestUtil.java [hunk 4]
```diff
@@ -21,4 +21,5 @@
      * @see <a href="https://www.testcontainers.org">https://www.testcontainers.org</a>
      */
     public static final String DATABASE_TESTCONTAINER = "spring.datasource.url=jdbc:tc:postgresql:11.13-alpine:///prs";
+
```
Comment (cpeeyush) [inline coreservices/partsrelationshipservice/prs-testing/src/main/java/net/catenax/prs/testing/TestUtil.java:24]:
```suggestion
```

File: coreservices/partsrelationshipservice/broker-proxy-integration-tests/src/test/java/net/catenax/brokerProxy/UpdatePartAspectTest.java [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+import java.util.List;
+
+import static io.restassured.RestAssured.given;
+import static net.javacrumbs.jsonunit.assertj.JsonAssertions.assertThatJson;
+import static net.javacrumbs.jsonunit.core.Option.IGNORING_ARRAY_ORDER;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class UpdatePartAspectTest extends BrokerProxyIntegrationTestBase {
+
+    private static final String PATH = "/brokerproxy/v0.1/PartAspectUpdate";
```
Comment (cpeeyush) [inline coreservices/partsrelationshipservice/broker-proxy-integration-tests/src/test/java/net/catenax/brokerProxy/UpdatePartAspectTest.java:21]:
```suggestion
    private static final String PATH = "/brokerproxy/v0.1/partAspectUpdate";
```

File: coreservices/partsrelationshipservice/broker-proxy-integration-tests/src/test/java/net/catenax/brokerProxy/UpdatePartRelationshipTest.java [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+import java.util.List;
+
+import static io.restassured.RestAssured.given;
+import static net.javacrumbs.jsonunit.assertj.JsonAssertions.assertThatJson;
+import static net.javacrumbs.jsonunit.core.Option.IGNORING_ARRAY_ORDER;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class UpdatePartRelationshipTest extends BrokerProxyIntegrationTestBase {
+
+    private static final String PATH = "/brokerproxy/v0.1/PartRelationshipUpdateList";
```
Comment (cpeeyush) [inline coreservices/partsrelationshipservice/broker-proxy-integration-tests/src/test/java/net/catenax/brokerProxy/UpdatePartRelationshipTest.java:19]:
```suggestion
    private static final String PATH = "/brokerproxy/v0.1/partRelationshipUpdateList";
```

File: coreservices/partsrelationshipservice/broker-proxy-integration-tests/src/test/java/net/catenax/brokerProxy/UpdatePartsAttributesTest.java [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+import static io.restassured.RestAssured.request;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import static io.restassured.RestAssured.given;
+import static net.javacrumbs.jsonunit.assertj.JsonAssertions.assertThatJson;
+import static net.javacrumbs.jsonunit.core.Option.IGNORING_ARRAY_ORDER;
+
+public class UpdatePartsAttributesTest extends BrokerProxyIntegrationTestBase {
+
+    private static final String PATH = "/brokerproxy/v0.1/PartAttributeUpdate";
```
Comment (cpeeyush) [inline coreservices/partsrelationshipservice/broker-proxy-integration-tests/src/test/java/net/catenax/brokerProxy/UpdatePartsAttributesTest.java:30]:
```suggestion
    private static final String PATH = "/brokerproxy/v0.1/partAttributeUpdate";
```

File: coreservices/partsrelationshipservice/broker-proxy-integration-tests/pom.xml [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+
+    <groupId>net.catenax.prs</groupId>
+    <artifactId>broker-proxy-integration-tests</artifactId>
+    <version>0.0.1-SNAPSHOT</version>
+
+    <name>Broker Api Tests</name>
+    <description>Broker Service Integration Tests</description>
+
+    <properties>
+        <groups>BrokerApiIntegrationTests</groups>
```
Comment (cpeeyush) [inline coreservices/partsrelationshipservice/broker-proxy-integration-tests/pom.xml:20]:
```suggestion
        <groups>IntegrationTests</groups>
```

File: coreservices/partsrelationshipservice/broker-proxy-integration-tests/src/test/java/net/catenax/brokerProxy/UpdatePartRelationshipTest.java (hunk 6 above)
Comment (Izzzu) [reply coreservices/partsrelationshipservice/broker-proxy-integration-tests/src/test/java/net/catenax/brokerProxy/UpdatePartRelationshipTest.java]: fixed

File: coreservices/partsrelationshipservice/broker-proxy/src/main/java/net/catenax/brokerproxy/requests/PartAttributeUpdateRequest.java [hunk 9]
```diff
@@ -37,7 +37,7 @@
     private PartId part;
 
     @NotNull
-    @ValueOfEnum(enumClass = PartAttribute.class)
+    @ValueOfEnum(enumClass = PartAttribute.class, message = "Invalid attribute name.")
     @Schema(description = "Attribute name")
```
Comment (cpeeyush) [inline coreservices/partsrelationshipservice/broker-proxy/src/main/java/net/catenax/brokerproxy/requests/PartAttributeUpdateRequest.java:41]:
```suggestion
     @Schema(implementation = PartAttribute.class, description = "Attribute name")
```

**Pull request merged**